lib/modify_game_state.rb

```diff
@@ -10,6 +10,6 @@ def execute(tile, player)
     if(new_board[tile-1] == 0)
       new_board[tile-1] = player
     end
-    return new_board
+    @game_state_gateway.save(new_board)
   end
```

Comment:
Be careful of implicit returns in Ruby - this method actually still returns the internal board. Typically I write a unit test to assert that use cases like this return `{}` e.g. `expect(response).to eq({})`

Reply:
Will bear this in mind and create a seperate pull request to fix this issue - possibly just returning a boolean to indicate the method was successfully executed